### DISCUSSION TITLE
[Chapter 08] @ManyToOne 관련 테스트 이슈

### DISCUSSION THREAD

**Discussion by Youngju-Jang** [cowaine/spring-boot-for-msa#123] [Chapter 08. Data Persistence]
559 페이지의 testCreateHotel 를 실행하던 중에
기존에 `HotelRoomEntity`에 `ManyToOne` HotelEntity 가 있으면 테스트 통과가 되지않는 문제 발생

![image](https://github.com/cowaine/spring-boot-for-msa/assets/77484084/21dd0635-2f91-4006-86fa-6ec4ff478cdb)

아래와 같이 `Unknown column 'hotel_entity_hotel_id' in 'field list'` 에러가 뜹니다.
`HotelRooms`테이블에 'hotel_entity_hotel_id' 라는 컬럼으로 설정되기에 `hotelEntity`필드가 문제인것 같아 주석처리후 실행하니 
![image](https://github.com/cowaine/spring-boot-for-msa/assets/77484084/38166c70-85d1-49f0-9dfe-ae65cea4b361)

위와같이 잘 실행되고, `hotel_entity_hotel_id` 컬럼에 insert하는 부분도 사라졌습니다.

```java
//     @ManyToOne
//     private HotelEntity hotelEntity;
```
위 내용에 @Column 태그가 없는데도 컬럼으로 인식하고 넣으려 해서 생긴 문제일까요?
562p 부터 다시 위에 코드를 넣는부분이 나오는데 곤란하네요

fc19062d8ff0c08bf9634dfd6b54bfabfc9af2f9

**Comment by Youngju-Jang** [comment]
다음페이지에서 `@JoinColumn(name = "hotel_id", nullable = true)` 추가하는 내용이 나오는데
![image](https://github.com/cowaine/spring-boot-for-msa/assets/77484084/a4857733-abee-43f0-9ce1-736101d2de9a)

라는 에러가 뜨는 중입니다.
왠지 저 `JoinColumn`내용 추가하면 해결될것 같은데 에러ㅠㅠㅠ 시간이없어서 해결을 마저 못하고 갑니다..ㅠㅠ 도움요청..ㅠㅠㅠ
b8d2b6de7c13e3c45a36beab47c10744da8fbade

**Comment by JoisFe** [comment]
일대다 매핑, 단방향 관계 테스트를 위해
HotelEntity에

``` java
   @OneToMany(cascade = CascadeType.PERSIST)
    @JoinColumn(name = "hotels_hotel_id", referencedColumnName = "hotel_id")
    private List<HotelRoomEntity> hotelRoomEntities;
```

이걸 넣어두셨는데 그 상황에서 `@ManyToOne` 을 추가하니 양방향 관계가 맺어지게된 것 으로 보입니다.
만약 양방향관계로 테스트하는게 목적이셨다면, mappedBy로 연관관계 주인을 지정해주셔야 하지 않을까요?
